layers/dns.go

```diff
@@ -370,6 +373,12 @@ func (d *DNS) DecodeFromBytes(data []byte, df gopacket.DecodeFeedback) error {
 			d.Additionals = d.Additionals[:i] // strip off erroneous value
 			return err
 		}
+		// extract extended RCODE from OPT RRs, RFC 6891 section 6.1.3
+		if d.Additionals[i].Type == DNSTypeOPT {
+			ttldata := make([]byte, 4)
+			binary.BigEndian.PutUint32(ttldata[:], d.Additionals[i].TTL)
+			d.ResponseCode = DNSResponseCode(uint8(d.ResponseCode) | uint8(ttldata[0])<<4)
```

Comment:
Isn't there some better way to do that than to allocate memory? I think some simple bit arithmetic should work (TTL will be in host byte order at this point - so if I'm not mistaken `d.Additionals[i].TTL>>24` and then `<<4` or shift by 16 and mask out lower 4 bits)